cluster/validate-cluster.sh

```diff
@@ -57,35 +61,35 @@ while true; do
     sleep 15
   fi
 done
-echo "Found ${found} nodes."
-echo -n "        "
-head -n 1 "${MINIONS_FILE}"
-tail -n +2 "${MINIONS_FILE}" | cat -n
+echo "Found ${found} node(s)."
+"${KUBE_ROOT}/cluster/kubectl.sh" get nodes
 
 attempt=0
 while true; do
-  kubectl_output=$("${KUBE_ROOT}/cluster/kubectl.sh" get cs) || true
+  kubectl_output=$("${KUBE_ROOT}/cluster/kubectl.sh" get cs --api-version="${API_VERSION}" -o yaml) || true
 
-  # The "kubectl componentstatuses" output is four columns like this:
+  # The "kubectl componentstatuses -o yaml" exports components health information in standard yaml format
   #
-  #     COMPONENT            HEALTH    MSG       ERR
-  #     controller-manager   Healthy   ok        nil
-  #
-  # Parse the output to capture the value of the second column("HEALTH"), then use grep to
-  # count the number of times it doesn't match "Healthy".
-  non_success_count=$(echo "${kubectl_output}" | \
-    sed '1d' |
-    sed -n 's/^[[:alnum:][:punct:]]/&/p' | \
-    grep --invert-match -c '^[[:alnum:][:punct:]]\{1,\}[[:space:]]\{1,\}Healthy') || true
+  # Echo the output and gather 2 counts:
+  #  - Total number of componentstatuses.
+  #  - Number of "healthy" components.
+  
+  componentstatuses=1;healthy=0
+  if [[ $API_VERSION == v1 ]]; then
+    # when there is a 'kind: ComponentStatus' line, there is a ComponentStatus found.
+    componentstatuses=$(echo "${kubectl_output}" | grep -c 'kind: ComponentStatus') || true
+    # only when status: "True" and type: Healthy both exist in conditions section, the compoment is "healthy".
+    healthy=$(echo "${kubectl_output}" | sed -n 'N;/status: "True".*type: Healthy/p; D;' | grep -c 'True') || true
+  fi
 
-  if ((non_success_count > 0)); then
+  if ((componentstatuses > healthy)); then
```

Comment:
There's a value initial of `componentstatuses` and `healthy`. It will not pass the check unless these two values are handled properly for each version. Currently for v1, it works. For the coming v2 or else, we need to define a different handle process for them other than for v1.